### PR TITLE
Add a 'delete_tags' cfp task

### PR DESCRIPTION
**Pull request by SamLR**
Because these will change and this way we can clean up

### CODE REVIEW

File: apps/cfp/tasks.py [hunk 1]
```diff
@@ -150,3 +150,31 @@ def create_tags(tags_to_create):
...
     db.session.commit()
     app.logger.info(f"Successfully created {tags_created} new tags.")
+
+
+@cfp.cli.command(
+    "delete_tags",
+    help=f"Delete tags from the Database. Any tagged proposals will have the tags removed.",
+)
+@click.argument("tags_to_delete", required=True, nargs=-1)
+def Delete_tags(tags_to_delete):
```
Comment (lukegb) [inline apps/cfp/tasks.py:160]: drive-by nit

```suggestion
def delete_tags(tags_to_delete):
```

File: apps/cfp/tasks.py [hunk 2]
```diff
@@ -150,3 +150,31 @@ def create_tags(tags_to_create):
...
+            app.logger.info(
+                f"'{tag_name}' is still applied to the following proposals: {tagged_proposals}"
+            )
+
+        db.session.delete(tag)
+        tags_deleted += 1
+        app.logger.info(f"'{tag_name}' added to session.")
+
+    db.session.commit()
+    app.logger.info(f"Successfully created {tags_deleted} new tags.")
```
Comment (lukegb) [inline apps/cfp/tasks.py:180]:
```suggestion
    app.logger.info(f"Successfully deleted {tags_deleted} tags.")
```

File: apps/cfp/tasks.py [hunk 3]
```diff
@@ -150,3 +150,31 @@ def create_tags(tags_to_create):
 
     db.session.commit()
     app.logger.info(f"Successfully created {tags_created} new tags.")
+
+
+@cfp.cli.command(
+    "delete_tags",
+    help=f"Delete tags from the Database. Any tagged proposals will have the tags removed.",
```
Comment (marksteward) [inline apps/cfp/tasks.py:157]: I guess "Any tagged proposals will have the tags removed" was a first draft?

File: apps/cfp/tasks.py (hunk 1 above)
Comment (SamLR) [reply apps/cfp/tasks.py]: not sure what happened there

File: apps/cfp/tasks.py (hunk 2 above)
Comment (SamLR) [reply apps/cfp/tasks.py]: done

File: apps/cfp/tasks.py (hunk 3 above)
Comment (SamLR) [reply apps/cfp/tasks.py]: no, the tags have got the correct cascade setting so deleting the tag removes it from the list (at least that's what local testing did)

Comment (marksteward) [reply apps/cfp/tasks.py]: Ahh, I read a `return` that isn't there. Maybe it should say something like f"'{tag_name}' will be removed from the following proposals: {tagged_proposals}".

**Review by lukegb — APPROVED** [review]
LGTMBWDIK (looks good to me, but what do I know)

File: apps/cfp/tasks.py (hunk 3 above)
Comment (SamLR) [reply apps/cfp/tasks.py]: Good point, :172 is now? `f"'{tag_name}' will be removed from the proposals with ids: {tagged_proposals}"`

File: apps/cfp/tasks.py [hunk 4]
```diff
@@ -150,3 +150,31 @@ def create_tags(tags_to_create):
...
+    for tag_name in tags_to_delete:
+        tag = Tag.query.filter_by(tag=tag_name).one_or_none()
+        if not tag:
+            app.logger.info(f"Couldn't find tag: '{tag_name}' exiting.")
+            return
+
+        if tag.proposals:
+            tagged_proposals = [p.id for p in tag.proposals]
+            app.logger.info(
+                f"'{tag_name}' will be removed from the proposals with ids: {tagged_proposals}"
```
Comment (jayaddison) [inline apps/cfp/tasks.py:172]: Nitpicky: would it be helpful to mention each proposal `title` as well?  (thinking: could provide a useful sense-check during tag removal)

Comment (SamLR) [reply apps/cfp/tasks.py:172]: probably not? This will hopefully be a pretty rare manual task (after I do what I need to do which is update the current list of tags to match those selected by the content team) and untagged proposals can be found via the UI. 

In theory deleting a tag will only happen when it's not useful: i.e. has very few (probably no) members OR so many members it's pointless.

Comment (jayaddison) [reply apps/cfp/tasks.py:172]: Okie doke, just checking.  Seems reasonable!
